File: hello.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include "plus.h"

void run(int count)
{
    long long start = current_timestamp();
    long long end;
    
    int x = 0;
    while (x < count)
        x = plusone(x);
    
    printf("result=%d, time=%lld\n", x, current_timestamp() - start);
}

int main(int argc, char** argv)
{
    if (argc == 1) {
        printf("First arg is required \n");
        return 1;
    }
    
    int count = atoi(argv[1]);
    if (count <= 0 || count > 2000000000) {
        printf("Must be a positive number not exceeding 2 billion.\n");
        return 1;
    }
    
    // start immediately
    run(count);

    return 0;
}

File: hello.py
import sys
from hello_cython import run

def main():

    count = int(sys.argv[1])
    if count <=0 or count > 2000000000:
        print("Must be a positive number not exceeding 2 billion.")
        return 1

    run(count)
    return 0

if __name__ == "__main__":
    main()
